mmengine/optim/optimizer/amp_optimizer_wrapper.py

```diff
@@ -68,6 +68,19 @@ def __init__(self, loss_scale='dynamic', **kwargs):
             raise TypeError('loss_scale must be of type float, dict, or '
                             f'"dynamic", but got {loss_scale}')
 
+        # convert string value to torch.dtype
+        if isinstance(dtype, str):
+            try:
+                dtype = getattr(torch, dtype)
+            except AttributeError:
+                raise ValueError(f'{dtype} does not name a torch.dtype. '
+                                 'You may want to use "float32", '
+                                 '"bfloat32" etc.')
```

Comment:
"float16" or "bfloat16"?

Reply:
Since this is an optimizer wrapper for mixed-precision training, should we control the `dtype` in 'float16' and 'bfloat16'?

Reply:
> "float16" or "bfloat16"?

Yes, a typo, will be fixed soon

> Since this is an optimizer wrapper for mixed-precision training, should we control the `dtype` in 'float16' and 'bfloat16'?

This is necessary because default `dtype` in `torch.cuda.autocast` is `torch.float16`. However, many consider `torch.bfloat16` better because it has more exponential digits. We should provide users with more options.

Reply:
oh, sorry for my poor expression. I mean could we just do the check like this:
```python
if isinstance(dtype, str):
   assert dtype in ('float16', 'bfloat16')
```
Then we do not need to use `try...except` anymore.

Reply:
Reasonable